File: letcode/problems/01_Two Sum/resolve.ts
const twoSum = (nums: Array<number>, target: number) => {
  const result: any = []
  outerLoop:
  for (let i = 0; i <= nums.length - 1; i++) {
    for (let a = i + 1; a <= nums.length - 1; a++) {
      let value = nums[i] + nums[a]
      if (value === target) {
        result.push(i, a)
        break outerLoop;
      }
    }
  }
}

const twoSum2 = (nums: Array<number>, target: number) => {
  const findIndex = (i: number, a: number): Array<number> => {
    if (i >= nums.length - 1) {
      return [];
    }

    if (a >= nums.length) {
      return twoSum2(nums.slice(i + 1), target);
    }

    if (nums[i] + nums[a] === target) {
      return [i, a];
    }
    return findIndex(i, a + 1);
  };

  return findIndex(0, 1);
}

twoSum2([2, 5, 5, 11], 10)

File: letcode/problems/01_Two Sum/Resolve.java
class Resolve {
  public static void main(String[] args) {
    Resolve resolve = new Resolve();

    int[] nums = {2, 7, 11, 15};
    int target = 9;

    System.out.println(resolve.twoSum(nums, target));
  }

  public int[] twoSum(int[] nums, int target) {
    for (int i = 0; i < nums.length; i++) {
      for (int j = i + 1; j < nums.length; j++) {
        if (nums[i] + nums[j] == target) {
          return new int[] { i, j };
        }
      }
    }
    return nums;
  }
}

/**
 * Given [1,2,3,5] -> 7
 * Result [2,5] -> [1,3] || [3,1]
 */
